### ISSUE TITLE
Getting silly results with field^-1

### ISSUE THREAD

**Issue by ornicar** [sksamuel/elastic4s#1148]
- elastic4s 5.6.0
- elasticsearch 6.0
- scala 2.12.4

When calling `fields("foo")` without specifying a boost, elastic4s sends `foo^-1.0` to elasticsearch. Instead of omitting the boost by just sending `foo`, it adds a default boost of `-1.0`.

Maybe that worked well before elasticsearch 6.0, I don't know. What I know is that with elasticsearch 6.0, `foo` and `foo^-1.0` return very different results. It seems that the query term will actually be avoided when the boost is `-1.0`. Which kind of makes sense.

Here's my query generated by elastic4s:
```json
                "multi_match":{
                  "query":"dragon",
                  "fields":[
                      "chapterNames^-1.0",
                      "chapterTexts^-1.0",
                      "members^-1.0",
                      "name^-1.0",
                      "owner^-1.0"
                  ],
                  "type":"most_fields",
                  "operator":"OR",
                  "analyzer":"english",
                  "slop":0,
                  "prefix_length":0,
                  "max_expansions":50,
                  "lenient":false
                  "zero_terms_query":"NONE",
                  "boost":1.0
                }
```

It returns results that never contain the keyword "dragon". When I fix the query by removing the `^-1.0`, like this:

```json
                "multi_match":{
                  "query":"dragon",
                  "fields":[
                      "chapterNames",
                      "chapterTexts",
                      "members",
                      "name",
                      "owner"
                  ],
                  "type":"most_fields",
                  "operator":"OR",
                  "analyzer":"english",
                  "slop":0,
                  "prefix_length":0,
                  "max_expansions":50,
                  "lenient":false
                  "zero_terms_query":"NONE",
                  "boost":1.0
                }
```
then I get correct results containing the "dragon" keyword.

**Comment by avdv** [comment]
I ran into the same issue using Elasticsearch 5.6.4.

**Comment by ornicar** [comment]
Thanks. Are we sure that no boost and a boost of 1 are treated equally by elasticsearch? Their doc doesn't mention it. An explicit boost of 1 could maybe cancel the mapping boost, for example.

**Comment by avdv** [comment]
> Thanks. Are we sure that no boost and a boost of 1 are treated equally by elasticsearch? Their doc doesn't mention it. An explicit boost of 1 could maybe cancel the mapping boost, for example.

Good point.

-1 is probably used as a way to encode a "default boost factor" throughout elastic4s's code base which is then removed before sending off the query (which just did not work [here](https://github.com/avdv/elastic4s/blob/f9fc7f0193254214e5d619b28ae40f007759b629/elastic4s-tcp/src/main/scala/com/sksamuel/elastic4s/searches/queries/QueryStringBuilderFn.scala#L18)). If that is the case, it would mean you could not use a -1 as a negative boost factor if you so desired; because that would effectively result in a default boost.

All in all, it probably would be better encoding that explicitly in the type instead of using some magic value.

@sksamuel could you shed some light?

**Comment by ornicar** [comment]
Yes it smells like `Option`, big times.

**Comment by sksamuel** [comment]
I did encode "no value" as -1d. Why? I was probably on smack or something.
The correct thing to do is encode it is an Option of course. There's a PR open which I think I should decline for 6.0 because it also uses a value rather than an Option.
I'll get this sorted before 6.0 final this weekend (bit later than I planned but my job got in the way).

**Comment by sksamuel** [comment]
Fixed in master for 6.0 release.
